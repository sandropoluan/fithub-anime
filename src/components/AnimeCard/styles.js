import styled from "styled-components";

export const Container = styled.div`
    padding: 8px;
    width: 200px;
    height: 340px;
    background-color: rgba(255,255,255, 0.5);
    margin: 0 8px 24px 8px;
    display: flex;
    flex-direction: column;
    align-items: center;
    border-radius: 10px;
    border: 2px #E4C8AE solid;
`;


export const Image = styled.img`
    height: 250px;
    border-radius: 5px 5px 0 0;
`;

export const Title = styled.span`
    width: 100%;
    margin-top: 8px;
    font-weight: 800;
    color:rgb(89, 104, 120);
    display: -webkit-box;
    -webkit-line-clamp: 2; 
    -webkit-box-orient: vertical;
    overflow: hidden;
    text-overflow: ellipsis;
    max-height: 3em; 
    line-height: 1.5em; 
`