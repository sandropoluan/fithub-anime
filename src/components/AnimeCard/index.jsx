import React, { useCallback } from "react";
import { useNavigate } from 'react-router';
import { Container, Durationtag, Image, RatingTag, Title, TypeTag } from "./styles";

export default function AnimeCard({data}){
    const { images, title, type, rating, duration, mal_id } = data;
    const { webp } = images;

    const navigate = useNavigate()

    const onClick = useCallback(() => {
        navigate(`/detail/${mal_id}`)
    }, [mal_id]);

    return <Container onClick={onClick}>
        {type && <TypeTag>{type}</TypeTag>}
        <RatingTag>{rating}</RatingTag>
        <Durationtag>{duration}</Durationtag>
        <Image role="img" src={webp.image_url}/>
        <Title>{title}</Title>
    </Container>
}