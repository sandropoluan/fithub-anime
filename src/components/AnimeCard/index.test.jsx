import React from 'react';
import { render, screen } from '@testing-library/react';
import AnimeCard from './index';
import '@testing-library/jest-dom';

const mockData = {
    images: {
        webp: {
            image_url: 'https://example.com/image.jpg'
        }
    },
    title: 'Anime Title',
    type: 'TV',
    rating: 'PG-13',
    duration: '24 min per ep',
    mal_id: 1
};

test('AnimeCard matches snapshot', () => {
    const { asFragment } = render(<AnimeCard data={mockData} />);
    expect(asFragment()).toMatchSnapshot();
});

test('AnimeCard renders Title', () => {
    render(<AnimeCard data={mockData} />);
    const titleElement = screen.getByText(/Anime Title/i);
    expect(titleElement).toBeInTheDocument();
});


test('AnimeCard renders type tags', () => {
    render(<AnimeCard data={mockData} />);
    const typeElement = screen.getByText(/TV/i);
    expect(typeElement).toBeInTheDocument();
});

test('AnimeCard renders rating tag', () => {
    render(<AnimeCard data={mockData} />);
    const ratingElement = screen.getByText(/PG-13/i);
    expect(ratingElement).toBeInTheDocument();
});

test('AnimeCard renders duration tag', () => {
    render(<AnimeCard data={mockData} />);
    const durationElement = screen.getByText(/24 min per ep/i);
    expect(durationElement).toBeInTheDocument();
});

test('AnimeCard renders image with correct src', () => {
    render(<AnimeCard data={mockData} />);
    const imageElement = screen.getByRole('img');
    expect(imageElement).toHaveAttribute('src', 'https://example.com/image.jpg');
});
