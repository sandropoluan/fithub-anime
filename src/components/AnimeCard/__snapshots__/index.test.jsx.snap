// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-FEMpB gPlbnh"
  >
    <span
      class="sc-fLDLck hEZIaG"
    >
      TV
    </span>
    <span
      class="sc-dycHyt eWkOvD"
    >
      PG-13
    </span>
    <span
      class="sc-jbAkgO hA-DFUm"
    >
      24 min per ep
    </span>
    <div
      class="sc-gjcoXW gSlUhQ"
      role="img"
      src="https://example.com/image.jpg"
    />
    <span
      class="sc-dVBluf iInBLV"
    >
      Anime Title
    </span>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-FEMpB gPlbnh"
  >
    <span
      class="sc-fLDLck hEZIaG"
    >
      TV
    </span>
    <span
      class="sc-dycHyt eWkOvD"
    >
      PG-13
    </span>
    <span
      class="sc-jbAkgO hA-DFUm"
    >
      24 min per ep
    </span>
    <div
      class="sc-gjcoXW gSlUhQ"
      role="img"
      src="https://example.com/image.jpg"
    />
    <span
      class="sc-dVBluf iInBLV"
    >
      Anime Title
    </span>
  </div>
</DocumentFragment>
`;
